doc/user/content/ingest-data/upsert-on-disk.md

```diff
@@ -0,0 +1,214 @@
+---
+title: "Upsert and Debezium on DISK Replicas"
+description: "How to size and manage large UPSERT and DEBEZIUM sources using DISK replicas."
+menu:
+  main:
+    parent: "kafka"
+    name: "Upsert and DEBEZIUM on `DISK` Replicas"
+    weight: 11
+---
+
+{{< note >}}
+This guide discusses a feature (namely, `DISK` replicas), that is currently only available as a preview. Please reach out to your Materialize
+representative for more information.
+{{</ note >}}
+
+This guide goes through how to run and manage large `UPSERT` or `DEBEZIUM` Kafka sources on `DISK` replicas.
+
+### Overview
+
+Kafka sources that use `ENVELOPE UPSERT` or `ENVELOPE DEBEZIUM` require the entire _working set_ to be stored somewhere, so they
+can produce retractions when keys are updated. The _working set_ is sum total of _current_ values for
+
+By default, Materialize uses an in-memory map to store the _working set_. However, sources that are scheduled to run on clusters whose
+replicas are configured to have an attached _disk_ (in reality its an ssd, but we call it a disk) using [_RocksDB_](https://rocksdb.org/).
+
+Using `DISK` replicas for these sources can reduce memory usage, and allow sources to be scheduled on smaller and cheaper replicas.
+
+### Sizing
+
+Materialize tries to ensure that any set of sources whose _working set_'s fit on a replica's attached disk can successfully run, but there are 2
+important parts of a source lifecycle that use additional resources, where care must be
+
+#### Hydration
+
+The first is _hydration_, when Materialize the large backlog of Kafka messages as it catches up with the end of the topic. Kafka sometimes
+allows Materialize to consume messages faster than they can be arranged into a _working set_ in RocksDB. Instead, they will be buffered into
+memory. In this case, a larger replica may temporarily be required
+
+{{< note >}}
+In the future this buffering limitation may be removed.
+{{</ note >}}
+
+To choose an initial size for _hydration_, please run:
+```sql
+select size, memory_bytes * processes / 1024 / 1024 / 1024 as disk_size_gib from mz_internal.mz_cluster_replica_sizes;
+```
+and choose the smallest size is _larger_ than your _estimated working set size_.
+
+#### Rehydration
+
+The second case is _rehydration_, when Materialize repopulates the _working set_ state in the replica when the replica is restarted.
+This typically only happens during maintenance windows.
+
+_Rehydration_ uses additional cpu and memory resources than steady-state processing. This is primarily due to processing within RocksDB itself,
+as well as (bounded) buffering of data being processed. It's important to ensure that your chosen replica size is sufficient for _rehydration_,
+and the below guide goes through this!
+
+### Guide
+
+1. #### Choose a Kafka guide
```

Comment:
nit: do you want to remove the numbers or make it part of the header? So `#### 1. Choose a Kafka guide` instead.

When viewing the file it's looking a bit weird with sudden indentations only for these headers (https://github.com/MaterializeInc/materialize/blob/2ab0964216935aebcab23047b76f149bceed794e/doc/user/content/ingest-data/upsert-on-disk.md#problems-during-hydration)

Reply:
nvm. I just realized we are going to format these doc pages, so it will not actually look like the github md preview here.